org.jacoco.core/src/org/jacoco/core/tools/ExecDumpClient.java

```diff
@@ -123,7 +123,10 @@ public ExecFileLoader dump(final InetAddress address, final int port)
 					.setExecutionDataVisitor(loader.getExecutionDataStore());
 
 			remoteWriter.visitDumpCommand(dump, reset);
-			remoteReader.read();
+
+			if (!remoteReader.read()) {
+				throw new IOException("Socket closed unexpectedly.");
```

Comment:
@Godin Yep, it's unrelated. I came across the following issue when writing test cases for cli: If a agent closes the connection without answering the the dump request we don't get an exception. The corresponding test case should be part of this PR: 

https://github.com/jacoco/jacoco/pull/525/files#diff-1b87cb3f6fcae3ed6f0c4bc8cd4fbe6c

If you want I can revert the change here and create a separate PR? The fix obviously applies to Ant ant Maven too.